tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/DebugTestRunner.java

```diff
@@ -575,4 +626,12 @@ public static void destroy() {
     public BalServer getBalServer() {
         return balServer;
     }
+
+    /**
+     * Debugger test framework supports both modes (hard assertions and soft assertions).
+    */
+    public enum AssertionMode {
+        HARD_ASSERT,
+        SOFT_ASSERT,
+    }
```

Comment:
Why do we need to introduce AssertionMode? Why do we need soft assert here?

Reply:
@praveennadarajah Yes please refer https://github.com/ballerina-platform/ballerina-lang/pull/27457/commits/23fdb18d98ab5fd5a6a5b984ddbc2f560dede2d8. Added this soft assertion mode to see all the assertion failures at once, instead of immediately failing on a single assertion error. (This mode is useful specially when fixing test failures). Anyway the tests are running in hard assertion mode by default.

Reply:
+1 for having soft assert.